numpy/core/src/multiarray/descriptor.c

```diff
@@ -1215,14 +1209,14 @@ _convert_from_dict(PyObject *obj, int align)
     }
     new->elsize = totalsize;
     if (!PyTuple_Check(names)) {
-        names = PySequence_Tuple(names);
-    }
-    else {
-        Py_INCREF(names);
+        Py_SETREF(names, PySequence_Tuple(names));
```

Comment:
Needs a null check after the re-assignment - `PySequence_Tuple` might have failed.

Reply:
fixing